src/main/scala/fs2/Process1.scala

```diff
@@ -104,6 +104,12 @@ object process1 {
   def scan1[F[_],I](f: (I, I) => I)(implicit F: NotNothing[F]): Handle[F,I] => Pull[F,I,Handle[F,I]] =
     Pull.receive1 { case o #: h => scan(o)(f).apply(h) }
 
+  /** Writes the sum of all input elements, or zero if the input is empty. */
+  def sum[F[_],I:Numeric](implicit F: NotNothing[F]): Handle[F,I] => Pull[F,I,Handle[F,I]] = {
```

Comment:
I don't really want to add `Monoid` to FS2 at this time. The policy is -
the util pkg should be the absolute bare minimum needed to make the library
useful. Monad, etc needs to be there because otherwise we'd have to pick a
specific effect type, which would defeat the whole purpose of the library,
which is to be able to work with effectful streams of any sort.

If there were a Monoid type in FS2, users would have to create instances.
But by the time you give an FS2 Monoid instance you could have just as
easily (with less code, actually) just called fold, scan, etc. It's just
not a big enough win to justify adding to this library.

Stuff that mentions a `Monoid` type belongs in a separate library that
depends on FS2 and whatever other library comes with a `Monoid` type.

On Tue, Nov 3, 2015 at 9:23 PM Jed Wesley-Smith <notifications@github.com>
wrote:

> In src/main/scala/fs2/Process1.scala
> <https://github.com/scalaz/scalaz-stream/pull/473#discussion_r43836112>:
>
> > @@ -104,6 +104,12 @@ object process1 {
> >    def scan1[F[_],I](f: (I, I) => I)(implicit F: NotNothing[F]): Handle[F,I] => Pull[F,I,Handle[F,I]] =
> >      Pull.receive1 { case o #: h => scan(o)(f).apply(h) }
> >
> > +  /** Writes the sum of all input elements, or zero if the input is empty. */
> > +  def sum[F[_],I:Numeric](implicit F: NotNothing[F]): Handle[F,I] => Pull[F,I,Handle[F,I]] = {
>
> oh, there were a whole bunch of them, I hadn't seen the specialised sum
> for Numeric :-( was thinking more of fold, foldMonoid and foldMap
>
> fs will definitely need Monoid.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scalaz/scalaz-stream/pull/473/files#r43836112>.
>